### ISSUE TITLE
mutable Issue for `EventuallyConfigurationDefaults`

### ISSUE THREAD

**Issue by mirageoasis** [kotest/kotest#4147]
as mentioned here
#4018 

`EventuallyConfigurationDefaults` has some issues

1. since it's singleton object it's variables not thread safe and error prone if set to mutable
2. think it's visibility should change since default option should not change

because this issue is separate from #4018 
I will raise another pr after #4018 being merged
